resolvers/GroupChatResolver.ts

```diff
@@ -1,114 +1,132 @@
+				groupChats: [],
+				totalPages: 0,
+				pageNumber: 0
+			};
+		}
+		return {
+			groupChats,
+			totalPages: Math.ceil(totalCount / this.pageSize) - 1,
+			pageNumber: page
+		};
+	}
 
-  @Query(() => [GroupChat], { nullable: true })
-  async getGroupChatByStatus(@Arg("status") status: string) {
-    const GroupChat = await GroupChatModel.find({ status });
-    return GroupChat;
-  }
+	@Query(() => [ GroupChat ], { nullable: true })
+	async getGroupChatByStatus(@Arg('status') status: string) {
+		const GroupChat = await GroupChatModel.find({ status });
+		return GroupChat;
+	}
 
-  @Query(() => GroupChat, { nullable: true })
-  async getGroupChat(@Arg("id") id: string) {
-    const GroupChat = await GroupChatModel.findOne({ _id: id });
-    return GroupChat;
-  }
+	@Query(() => GroupChat, { nullable: true })
+	async getGroupChat(@Arg('id') id: string) {
+		const GroupChat = await GroupChatModel.findOne({ _id: id });
+		return GroupChat;
+	}
 
-  @Query(() => GroupChatPaginiated, { nullable: true })
-  async searchGroupChats(
-    @Arg("text", { nullable: true }) text?: string,
-    @Arg("isCommunity", { nullable: true }) type?: Boolean,
-    @Arg("page", { nullable: true }) page: number = 0
-  ) {
-    let queryObj = {};
-    if (text != undefined) {
-      const regex = new RegExp(escapeRegex(text), "gi");
-      queryObj = { description: regex };
-    }
-    if (type != undefined) {
-      queryObj = { ...queryObj, isCommunity: type };
-    }
-    const groupChats = await GroupChatModel.find(queryObj)
-      .skip(page * this.pageSize)
-      .limit(this.pageSize);
-    const totalCount = await GroupChatModel.find(queryObj).countDocuments();
-    if (totalCount === 0) {
-      return {
-        groupChats: [],
-        totalPages: 0,
-        pageNumber: 0,
-      };
-    }
-    return {
-      groupChats,
-      totalPages: Math.ceil(totalCount / this.pageSize) - 1,
-      pageNumber: page,
-    };
-  }
+	@Query(() => GroupChatPaginiated, { nullable: true })
+	async searchGroupChats(
+		@Arg('campus', { nullable: true })
+		campus?: string,
+		@Arg('department', { nullable: true })
+		department?: string,
+		@Arg('code', { nullable: true })
+		code?: string,
+		@Arg('term', { nullable: true })
+		term?: string,
+		@Arg('year', { nullable: true })
+		year?: string,
+		@Arg('text', { nullable: true })
+		text?: string,
+		@Arg('isCommunity', { nullable: true })
+		type?: Boolean,
+		@Arg('page', { nullable: true })
+		page: number = 0
+	) {
+		let queryObj = {};
+		if (campus != undefined) {
+			queryObj = { ...queryObj, courseInformation: { campus } };
+		}
+		if (department != undefined) {
+			queryObj = { ...queryObj, courseInformation: { department } };
+		}
+		if (code != undefined) {
+			queryObj = { ...queryObj, courseInformation: { code } };
+		}
+		if (term != undefined) {
+			queryObj = { ...queryObj, courseInformation: { term } };
+		}
+		if (year != undefined) {
+			queryObj = { ...queryObj, courseInformation: { year } };
+		}
```

Comment:
I believe you're re-assigning courseInformation here, which ignores the previous if statement assignments!

Reply:
We can go that route if you'd like! I have a habit of redefining rather than mutating, but that's just the functional side of me talking. Not sure about speed difference either, but if it helps clean it up, then that's great!